src/plugin.ts

```diff
@@ -63,12 +63,22 @@ export default {
 
 		// we need to share some logic between the plugin and the routeGuard
 		// maybe there is a better way to do it
-		routeGuardInternal.value = async () => {
-			if (!sessionToken.value && isFetchSessionWasNeverCalled.value) {
-				await fetchSession();
-			}
-			return !!unref(sessionToken);
-		};
+		routeGuardInternal.value = () =>
+			new Promise((resolve, reject) => {
+				if (!sessionToken.value && isFetchSessionWasNeverCalled.value) {
+					fetchSession().catch(reject);
```

Comment:
I think that this was the issue root cause, 

there was an edge case that the session has fetched, but the `isSessionLoading` is not updated yet